spyder/plugins/editor/widgets/completion.py

```diff
@@ -0,0 +1,401 @@
+# -*- coding: utf-8 -*-
+#
+# Copyright © Spyder Project Contributors
+# Licensed under the terms of the MIT License
+# (see spyder/__init__.py for details)
+
+"""Completion widget class"""
+
+# Standard library imports
+import sys
+
+# Third psrty imports
+from qtpy.QtCore import QPoint, Qt, Signal, Slot
+from qtpy.QtGui import QFontMetrics
+from qtpy.QtWidgets import (QAbstractItemView, QApplication, QListWidget,
+                            QListWidgetItem, QToolTip)
+
+# Local imports
+from spyder.utils import icon_manager as ima
+from spyder.plugins.completion.languageserver import CompletionItemKind
+from spyder.py3compat import to_text_string
+from spyder.widgets.helperwidgets import HTMLDelegate
+
+
+COMPLETION_ITEM_TEMPLATE = """
+<table width="{width}" height="{height}">
+    <tr>
+        <td valign="middle" style="color:{color}">
+            {completion}
+        </td>
+
+        <td valign="middle" align="right" float="right" style="color:{color}">
+            {type}
+        </td>
+    </tr>
+</table>
+"""
+DEFAULT_COMPLETION_ITEM_HEIGHT = 15
+DEFAULT_COMPLETION_ITEM_WIDTH = 250
+
+
+class CompletionWidget(QListWidget):
+    """Completion list widget"""
+
+    sig_show_completions = Signal(object)
+    sig_completion_hint = Signal(str, str, QPoint)
+
+    def __init__(self, parent, ancestor):
+        QListWidget.__init__(self, ancestor)
```

Comment:
Any reason we should use the class vs, super @ccordoba12 ?

(I saw that some issues with PySide2 were because of this)

Reply:
@ccordoba12 ?